### PR TITLE
[GUI Editor] Space + Click to Pan

### PR DESCRIPTION
**Pull request by darraghjburke**
Holding space in selection mode will now show the pan icon selected and will let you click and drag.

Under the hood, I've created a KeyboardManager class to track all modifier keys.

There is some major technical debt existing in all our tools; the lack of a state management system. We are relying on GlobalState and observables, but this makes it really easy to introduce bugs related to state changes. We may want to consider something like Redux in the future.

### CODE REVIEW

**Comment by darraghjburke** [comment]
Marking as draft as this may cause issues with the node editor due to the removal of keyboardBlocker. Going to investigate the right fix for this

**Comment by darraghjburke** [comment]
Will have to pick this up on Monday

**Comment by darraghjburke** [comment]
Ok, for now I have left the blockKeyboardEvents property in, in the interests of time. In the long run we will want to remove this variable from NME and GUI Editor, and use event.target instead to determine if keyboard shortcuts should act on the editor.

**Comment by RaananW** [comment]
@darraghjburke - just some conflicts (from the big PR before)

File: packages/tools/guiEditor/src/globalState.ts [hunk 1]
```diff
@@ -19,6 +20,12 @@ export enum DragOverLocation {
     NONE = 3,
 }
 
+export enum Tool {
```
Comment (RaananW) [inline packages/tools/guiEditor/src/globalState.ts:23]: I think "Tool" is a very abstract name for that. Maybe EditorTool, or GUIEditorControlTool, or something more specific?

**Review by RaananW — COMMENTED** [review]
Just a small comment about naming. otherwise LGTM!!

File: packages/tools/guiEditor/src/globalState.ts (hunk 1 above)
Comment (carolhmj) [reply packages/tools/guiEditor/src/globalState.ts:23]: +1

**Review by carolhmj — APPROVED** [review]
Agree with Raanan about the Tool naming, it's great otherwise! :)

**Comment by RaananW** [comment]
@darraghjburke - any update on this one?

File: packages/tools/guiEditor/src/globalState.ts (hunk 1 above)
Comment (darraghjburke) [reply packages/tools/guiEditor/src/globalState.ts:23]: Sure, we can do that

**Pull request merged**